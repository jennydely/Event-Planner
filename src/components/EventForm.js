import React, { useRef } from 'react'
import { useForm } from 'react-hook-form'
import styled from 'styled-components/macro'
import Input from '../common/Input'
import Label from '../common/Label'
import ErrorMessage from '../common/ErrorMessage'
import Select from '../common/Select'
import PropTypes from 'prop-types'

EventForm.propTypes = {
  onEventSave: PropTypes.func.isRequired,
  packlists: PropTypes.array.isRequired,
}

export default function EventForm({ onEventSave, packlists }) {
  const { register, handleSubmit, errors, reset } = useForm()
  const onSubmit = (eventEntry, event) => {
    // for testing...
    if (event && event.target && typeof event.target.reset === 'function')
      event.target.reset()
    onEventSave(eventEntry)
  }
  const endDateRef = useRef(null)
  const allPacklists = packlists.map((packlist) => packlist.name)
  return (
    <>
      <Form data-testid="1337" onSubmit={handleSubmit(onSubmit)}>
        <CategoryInputLabel htmlFor="category">Category:</CategoryInputLabel>
        <CategoryInput
          name="category"
          id="category"
          register={register({ required: true })}
          options={['metal', 'medieval', 'sand', 'other']}
        />
        {errors.category && errors.category.type === 'required' && (
          <ErrorMessageCategoryReq>
            Category is required!
          </ErrorMessageCategoryReq>
        )}

        <InputLabelColumn2 row={2} htmlFor="name">
          EventName:
        </InputLabelColumn2>
        <InputColumn2
          row={3}
          placeholder="event name"
          id="name"
          name="name"
          ref={register({
            required: true,
            minLength: 3,
            maxLength: 20,
            validate: (value) =>
              value && value.trim().length >= 3 && value.trim().length <= 20,
          })}
        />

        {errors.name?.type === 'required' && (
          <ErrorMessageColumn2 row={4}>Name is required!</ErrorMessageColumn2>
        )}
        {(errors.name?.type === 'validate' ||
          errors.name?.type === 'minLength') && (
          <ErrorMessageColumn2 row={4}>
            This field requires at least 3 characters!
          </ErrorMessageColumn2>
        )}
        {(errors.name?.type === 'validate' ||
          errors.name?.type === 'maxLength') && (
          <ErrorMessageColumn2 row={4}>
            The name can reach a maximum of 20 characters!
          </ErrorMessageColumn2>
        )}

        <InputLabelColumn2 row={5} htmlFor="location">
          EventLocation:
        </InputLabelColumn2>
        <InputColumn2
          row={6}
          placeholder="location of the event"
          id="location"
          name="location"
          ref={register({
            required: true,
            minLength: 3,
            maxLength: 20,
            validate: (value) =>
              value && value.trim().length >= 3 && value.trim().length <= 20,
          })}
        />
        {errors.location?.type === 'required' && (
          <ErrorMessageColumn2 row={7}>
            Location is required!
          </ErrorMessageColumn2>
        )}
        {(errors.location?.type === 'validate' ||
          errors.location?.type === 'minLength') && (
          <ErrorMessageColumn2 row={7}>
            {' '}
            This field requires at least 3 characters!
          </ErrorMessageColumn2>
        )}
        {(errors.location?.type === 'validate' ||
          errors.location?.type === 'maxLength') && (
          <ErrorMessageColumn2 row={7}>
            The location can reach a maximum of 20 characters!
          </ErrorMessageColumn2>
        )}

        <EventInfosText>EventInfos</EventInfosText>
        <InputLabelColumn1 row={9} htmlFor="EventStartDate">
          Duration:
        </InputLabelColumn1>

        <EventStartDateInput
          placeholder="yyyy-mm-dd"
          id="EventStartDate"
          name="eventStartDate"
          ref={register({
            required: true,
            pattern: /([12]\d{3}-(0[1-9]|1[0-2])-(0[1-9]|[12]\d|3[01]))/,
            validate: (value) => value <= endDateRef.current.value,
          })}
        />
        {errors.eventStartDate?.type === 'validate' && (
          <ErrorMessageStartDate>
            Start date must be before end date!{' '}
          </ErrorMessageStartDate>
        )}

        {errors.eventStartDate?.type === 'required' && (
          <ErrorMessageStartDate>
            Date is required and needs the right form!{' '}
          </ErrorMessageStartDate>
        )}

        {errors.eventStartDate?.type === 'pattern' && (
          <ErrorMessageStartDate>
            Date must be written like: yyyy-mm-dd
          </ErrorMessageStartDate>
        )}

        <EventEndDateInput
          placeholder="yyyy-mm-dd"
          id="EventEndDate"
          name="eventEndDate"
          ref={(el) => {
            endDateRef.current = el
            register(el, {
              required: true,
              pattern: /([12]\d{3}-(0[1-9]|1[0-2])-(0[1-9]|[12]\d|3[01]))/,
            })
          }}
        />

        {errors.eventEndDate?.type === 'required' && (
          <ErrorMessageEndDate>
            Date is required and needs the right form!{' '}
          </ErrorMessageEndDate>
        )}

        {errors.eventEndDate?.type === 'pattern' && (
          <ErrorMessageEndDate>
            {' '}
            Date must be written like: yyyy-mm-dd{' '}
          </ErrorMessageEndDate>
        )}

        <AddressInputLabel htmlFor="Street">Address:</AddressInputLabel>
        <InputColumn2
          row={11}
          placeholder="street + number"
          id="street"
          name="street"
          ref={register({
            required: true,
            minLength: 5,
            validate: (value) => value && value.trim().length >= 5,
          })}
        />

        {errors.street?.type === 'required' && (
          <ErrorMessageColumn2 row={12}>
            Street is required!
          </ErrorMessageColumn2>
        )}
        {(errors.street?.type === 'validate' ||
          errors.street?.type === 'minLength') && (
          <ErrorMessageColumn2 row={12}>
            This field requires at least 5 characters!
          </ErrorMessageColumn2>
        )}

        <InputColumn2
          row={13}
          placeholder="zip"
          id="zip"
          name="zip"
          ref={register({
            required: true,
            minLength: 2,
            validate: (value) => value && value.trim().length >= 2,
          })}
        />

        {errors.zip?.type === 'required' && (
          <ErrorMessageColumn2 row={14}>Zip is required!</ErrorMessageColumn2>
        )}
        {(errors.zip?.type === 'validate' ||
          errors.zip?.type === 'minLength') && (
          <ErrorMessageColumn2 row={14}>
            This field requires at least 2 characters!
          </ErrorMessageColumn2>
        )}

        <InputLabelColumn1 row={15} htmlFor="website">
          Website:
        </InputLabelColumn1>
        <InputColumn2
          row={15}
          placeholder="http://www.website.de"
          id="website"
          name="website"
          ref={register({
            required: true,
            minLength: 8,
            validate: (value) => value && value.trim().length >= 8,
            pattern: /(https?:\/\/(?:www\.|(?!www))[a-zA-Z0-9][a-zA-Z0-9-]+[a-zA-Z0-9]\.[^\s]{2,}|www\.[a-zA-Z0-9][a-zA-Z0-9-]+[a-zA-Z0-9]\.[^\s]{2,}|https?:\/\/(?:www\.|(?!www))[a-zA-Z0-9]+\.[^\s]{2,}|www\.[a-zA-Z0-9]+\.[^\s]{2,})/,
          })}
        />

        {errors.website?.type === 'required' && (
          <ErrorMessageColumn2 row={16}>
            Website is required!
          </ErrorMessageColumn2>
        )}
        {(errors.website?.type === 'validate' ||
          errors.website?.type === 'minLength') && (
          <ErrorMessageColumn2 row={16}>
            This field requires at least 8 characters!
          </ErrorMessageColumn2>
        )}
        {errors.website?.type === 'pattern' && (
          <ErrorMessageColumn2 row={16}>
            Should begin with www or http
          </ErrorMessageColumn2>
        )}

        <InputLabelColumn1 row={17} htmlFor="price">
          TicketPrice:
        </InputLabelColumn1>
        <InputColumn2
          row={17}
          placeholder="ticket price or range"
          id="price"
          name="price"
          ref={register({ required: false })}
        />
        <InputLabelColumn1 row={18} htmlFor="poster">
          Picture:{' '}
        </InputLabelColumn1>
        <InputColumn2
          row={18}
          placeholder="http://www.website.de/banner.jpg"
          id="poster"
          name="poster"
          ref={register({
            required: false,
            pattern: /(https?:\/\/(?:www\.|(?!www))[a-zA-Z0-9][a-zA-Z0-9-]+[a-zA-Z0-9]\.[^\s]{2,}|www\.[a-zA-Z0-9][a-zA-Z0-9-]+[a-zA-Z0-9]\.[^\s]{2,}|https?:\/\/(?:www\.|(?!www))[a-zA-Z0-9]+\.[^\s]{2,}|www\.[a-zA-Z0-9]+\.[^\s]{2,})/,
          })}
        />
        <InputLabelColumn1 row={19} htmlFor="packlist">
          Add PackList:
        </InputLabelColumn1>
        <PacklistInput
          name="packlistCategory"
          id="packlist"
          register={register({ required: true })}
          options={allPacklists}
        />
        <ButtonGroup>
          <button type="reset" onClick={() => reset()}>
            Reset
          </button>
          <SubmitButton type="submit">Save</SubmitButton>
        </ButtonGroup>
      </Form>
    </>
  )
}

const Form = styled.form`
  display: grid;
  grid-template-columns: repeat(3, auto);
  grid-template-rows: repeat(20, auto);
  align-content: center;
  min-width: 300px;
  gap: 4px;
`
const InputLabelColumn1 = styled(Label)`
  grid-column: 1;
  grid-row: ${(props) => props.row};
`
const InputLabelColumn2 = styled(Label)`
  grid-column: 2 / span 2;
  grid-row: ${(props) => props.row};
  text-align: left;
`
const InputColumn2 = styled(Input)`
  grid-column: 2 / span 2;
  grid-row: ${(props) => props.row};
`
const ErrorMessageColumn2 = styled(ErrorMessage)`
  grid-column: 2 / span 2;
  grid-row: ${(props) => props.row};
`
const CategoryInputLabel = styled(Label)`
  grid-column: 1;
  grid-row: 2;
  text-align: left;
`
const CategoryInput = styled(Select)`
  grid-column: 1;
  grid-row: 3;
  display: block;
  width: 100%;
  padding: 20px;
  border: 1px solid rgb(96, 99, 104);
  border-radius: 4px;
  margin-top: 0;
  padding: 4px;
  font-size: 112.5%;
  color: black;
  background: ${({ name }) =>
    name === 'sand'
      ? 'rgba(248,149,17,0.46)'
      : name === 'metal'
      ? 'rgba(49,42,42,0.75)'
      : name === 'medieval'
      ? 'rgba(67,40,24,0.70)'
      : name === 'other'
      ? 'rgba(153,88,42,0.70)'
      : 'rgb(96,99,104)'};
`
const ErrorMessageCategoryReq = styled(ErrorMessage)`
  grid-column: 1;
  grid-row: 4;
`
const EventInfosText = styled.h2`
  grid-column: 1 / span 2;
  grid-row: 8;
  margin-top: 20px;
`
const EventStartDateInput = styled(Input)`
  grid-column: 2;
  grid-row: 9;
`
const ErrorMessageStartDate = styled(ErrorMessage)`
  grid-column: 2;
  grid-row: 10;
`
const EventEndDateInput = styled(Input)`
  grid-column: 3;
  grid-row: 9;
`
const ErrorMessageEndDate = styled(ErrorMessage)`
  grid-column: 3;
  grid-row: 10;
`
const AddressInputLabel = styled(Label)`
  grid-column: 1;
  grid-row: 11 / span 2;
`
const PacklistInput = styled(Select)`
  grid-column: 2 / span 2;
  grid-row: 19;
  display: block;
  width: 100%;
  border-radius: 4px;
  border: 1px solid rgb(96, 99, 104);
  margin-top: 0;
  padding: 4px;
  font-size: 112.5%;
  color: black;
  background: rgb(96, 99, 104);
`
const ButtonGroup = styled.div`
  grid-column: 1 / span 3;
  grid-row: 20;
  display: flex;
  justify-content: center;
  width: 100%;
  margin: 20px 7px;
`
const SubmitButton = styled.button`
  grid-column: 1 / span 3;
  grid-row: 20;
  background-color: rgba(111, 29, 27, 0.75);
`
