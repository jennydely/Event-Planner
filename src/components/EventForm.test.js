import React from 'react'
import { render, fireEvent, waitFor } from '@testing-library/react'
import EventForm from './EventForm'
import 'jest-styled-components'

window.MutationObserver = require('mutation-observer')

describe('EventForm', () => {
  it('creates a new event', async () => {
    const onEventSave = jest.fn()
    const packlists = [
      {
        id: 500,
        name: 'festival',
        packlist: [
          'Wasser',
          'Plastikzelt',
          'Metalgürtel',
          'ausgedruckter Timetable',
        ],
      },
      {
        id: 501,
        name: 'medieval',
        packlist: [
          'Wasser',
          'Leinenzelt',
          'Bodenplane',
          'Schurwollmantel & Umhang',
        ],
      },
    ]
    const {
      getByText,
      getByLabelText,
      getByPlaceholderText,
      getAllByPlaceholderText,
    } = render(<EventForm packlists={packlists} onEventSave={onEventSave} />)

    fireEvent.change(getByLabelText('Category:'), {
      target: { value: 'medieval' },
    })
    fireEvent.input(getByPlaceholderText('event name'), {
      target: { value: 'My EventName' },
    })
    fireEvent.input(getByPlaceholderText('location of the event'), {
      target: { value: 'My EventLocation' },
    })
    fireEvent.input(getByPlaceholderText('street + number'), {
      target: { value: 'My Address' },
    })
    fireEvent.input(getByPlaceholderText('http://www.website.de'), {
      target: { value: 'www.website.de' },
    })
    fireEvent.input(getByPlaceholderText('zip'), { target: { value: '12345' } })
    fireEvent.input(getAllByPlaceholderText('yyyy-mm-dd')[0], {
      target: { value: '2020-05-20' },
    })
    fireEvent.input(getAllByPlaceholderText('yyyy-mm-dd')[1], {
      target: { value: '2020-05-25' },
    })

    fireEvent.submit(getByText('Save'))
    await waitFor(() => expect(onEventSave).toHaveBeenCalled())
  })

  it('displays a form to create events', async () => {
    const onEventSave = jest.fn()
    const packlists = [
      {
        id: 500,
        name: 'festival',
        packlist: [
          'Wasser',
          'Plastikzelt',
          'Metalgürtel',
          'ausgedruckter Timetable',
        ],
      },
      {
        id: 501,
        name: 'medieval',
        packlist: [
          'Wasser',
          'Leinenzelt',
          'Bodenplane',
          'Schurwollmantel & Umhang',
        ],
      },
    ]
    const { getByText, getByLabelText, getByPlaceholderText } = render(
      <EventForm packlists={packlists} onEventSave={onEventSave} />
    )

    const nameInput = getByPlaceholderText('event name')
    expect(nameInput.value).toBe('')
    fireEvent.input(nameInput, { target: { value: 'My EventName' } })
    expect(nameInput.value).toBe('My EventName')

    fireEvent.click(getByText('Reset'))
    await waitFor(() => expect(nameInput.value).toBe(''))
    expect(nameInput.value).toBe('')
    await waitFor(() => expect(getByLabelText('Category:').value).toBe('metal'))
  })

  it('renders correctly', () => {
    const onEventSave = jest.fn()
    const packlists = [
      {
        id: 500,
        name: 'festival',
        packlist: [
          'Wasser',
          'Plastikzelt',
          'Metalgürtel',
          'ausgedruckter Timetable',
        ],
      },
      {
        id: 501,
        name: 'medieval',
        packlist: [
          'Wasser',
          'Leinenzelt',
          'Bodenplane',
          'Schurwollmantel & Umhang',
        ],
      },
    ]
    const tree = render(
      <EventForm packlists={packlists} onEventSave={onEventSave} />
    )
    expect(tree).toMatchSnapshot()
  })
})
