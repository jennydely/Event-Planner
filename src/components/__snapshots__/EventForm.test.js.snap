// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventForm renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c5 {
  display: block;
  width: 100%;
  padding: 20px;
  border-radius: 4px;
  border: var(--border-darkgrey);
  margin-top: 0;
  padding: 4px;
  font-size: 112.5%;
  color: black;
}

.c1 {
  font-weight: 400;
  size: 112.5%;
  margin: 4px;
  text-align: right;
}

.c0 {
  display: grid;
  grid-template-columns: auto repeat(2,120px);
  grid-template-rows: repeat(20,auto);
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  min-width: 300px;
  gap: 4px;
}

.c10 {
  grid-column: 1;
  grid-row: 9;
}

.c16 {
  grid-column: 1;
  grid-row: 15;
}

.c18 {
  grid-column: 1;
  grid-row: 17;
}

.c20 {
  grid-column: 1;
  grid-row: 18;
}

.c22 {
  grid-column: 1;
  grid-row: 19;
}

.c4 {
  grid-column: 2 / span 2;
  grid-row: 2;
  text-align: left;
}

.c7 {
  grid-column: 2 / span 2;
  grid-row: 5;
  text-align: left;
}

.c6 {
  grid-column: 2 / span 2;
  grid-row: 3;
}

.c8 {
  grid-column: 2 / span 2;
  grid-row: 6;
}

.c14 {
  grid-column: 2 / span 2;
  grid-row: 11;
}

.c15 {
  grid-column: 2 / span 2;
  grid-row: 13;
}

.c17 {
  grid-column: 2 / span 2;
  grid-row: 15;
}

.c19 {
  grid-column: 2 / span 2;
  grid-row: 17;
}

.c21 {
  grid-column: 2 / span 2;
  grid-row: 18;
}

.c2 {
  grid-column: 1;
  grid-row: 2;
  text-align: left;
}

.c3 {
  grid-column: 1;
  grid-row: 3;
  display: block;
  width: 100%;
  padding: 20px;
  border: var(--border-darkgrey);
  border-radius: 4px;
  margin-top: 0;
  padding: 4px;
  font-size: 112.5%;
  color: black;
  background: var(--lightgrey-main);
}

.c9 {
  grid-column: 1 / span 2;
  grid-row: 8;
  margin-top: 20px;
}

.c11 {
  grid-column: 2/3;
  grid-row: 9;
}

.c12 {
  grid-column: 3;
  grid-row: 9;
}

.c13 {
  grid-column: 1;
  grid-row: 11 / span 2;
}

.c23 {
  grid-column: 2 / span 2;
  grid-row: 19;
  display: block;
  width: 100%;
  border-radius: 4px;
  border: var(--border-darkgrey);
  margin-top: 0;
  padding: 4px;
  font-size: 112.5%;
  color: black;
  background: var(--lightgrey-main);
}

.c24 {
  grid-column: 1 / span 3;
  grid-row: 20;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 100%;
  margin: 20px 7px;
}

.c25 {
  grid-column: 1 / span 3;
  grid-row: 20;
  background-color: var(--darkred-75);
}

<body>
    <div>
      <form
        class="c0"
        data-testid="eventform"
      >
        <label
          class="c1 c2"
          for="category"
        >
          Category:
        </label>
        <select
          class="c3"
          id="category"
          name="category"
        >
          <option
            value="metal"
          >
            metal
          </option>
          <option
            value="medieval"
          >
            medieval
          </option>
          <option
            value="holiday"
          >
            holiday
          </option>
          <option
            value="other"
          >
            other
          </option>
        </select>
        <label
          class="c1 c4"
          for="name"
        >
          EventName:
        </label>
        <input
          class="c5 c6"
          id="name"
          name="name"
          placeholder="event name"
        />
        <label
          class="c1 c7"
          for="location"
        >
          EventLocation:
        </label>
        <input
          class="c5 c8"
          id="location"
          name="location"
          placeholder="location of the event"
        />
        <h2
          class="c9"
        >
          EventInfos
        </h2>
        <label
          class="c1 c10"
          for="EventStartDate"
        >
          Duration:
        </label>
        <div
          class="c11"
        >
          <div
            class="react-datepicker-wrapper"
          >
            <div
              class="react-datepicker__input-container"
            >
              <input
                class="c5"
                placeholder="start date"
                value=""
              />
            </div>
          </div>
        </div>
        <div
          class="c12"
        >
          <div
            class="react-datepicker-wrapper"
          >
            <div
              class="react-datepicker__input-container"
            >
              <input
                class="c5"
                placeholder="end date"
                value=""
              />
            </div>
          </div>
        </div>
        <label
          class="c1 c13"
          for="Street"
        >
          Address:
        </label>
        <input
          class="c5 c14"
          id="street"
          name="street"
          placeholder="street + number"
        />
        <input
          class="c5 c15"
          id="zip"
          name="zip"
          placeholder="zip"
        />
        <label
          class="c1 c16"
          for="website"
        >
          Website:
        </label>
        <input
          class="c5 c17"
          id="website"
          name="website"
          placeholder="http://www.website.de"
        />
        <label
          class="c1 c18"
          for="price"
        >
          TicketPrice:
        </label>
        <input
          class="c5 c19"
          id="price"
          name="price"
          placeholder="ticket price or range"
        />
        <label
          class="c1 c20"
          for="poster"
        >
          Picture:
           
        </label>
        <input
          class="c5 c21"
          id="poster"
          name="poster"
          placeholder="http://www.website.de/banner.jpg"
        />
        <label
          class="c1 c22"
          for="packlist"
        >
          Add PackList:
        </label>
        <select
          class="c23"
          id="packlist"
          name="packlistCategory"
        >
          <option
            value="festival"
          >
            festival
          </option>
          <option
            value="medieval"
          >
            medieval
          </option>
        </select>
        <div
          class="c24"
        >
          <button
            type="reset"
          >
            Reset
          </button>
          <button
            class="c25"
            type="submit"
          >
            Save
          </button>
        </div>
      </form>
    </div>
  </body>,
  "container": <div>
    <form
      class="EventForm__Form-sc-1hr7ssh-0 druZya"
      data-testid="eventform"
    >
      <label
        class="Label-sc-4r18cf-0 EventForm__CategoryInputLabel-sc-1hr7ssh-5 kTDDAh kKWTAK"
        for="category"
      >
        Category:
      </label>
      <select
        class="EventForm__CategoryInput-sc-1hr7ssh-6 cezilq"
        id="category"
        name="category"
      >
        <option
          value="metal"
        >
          metal
        </option>
        <option
          value="medieval"
        >
          medieval
        </option>
        <option
          value="holiday"
        >
          holiday
        </option>
        <option
          value="other"
        >
          other
        </option>
      </select>
      <label
        class="Label-sc-4r18cf-0 EventForm__InputLabelColumn2-sc-1hr7ssh-2 kTDDAh bMMjQM"
        for="name"
      >
        EventName:
      </label>
      <input
        class="Input-sc-1qrq46f-0 EventForm__InputColumn2-sc-1hr7ssh-3 hKSNO ekNFAh"
        id="name"
        name="name"
        placeholder="event name"
      />
      <label
        class="Label-sc-4r18cf-0 EventForm__InputLabelColumn2-sc-1hr7ssh-2 kTDDAh eCVOzL"
        for="location"
      >
        EventLocation:
      </label>
      <input
        class="Input-sc-1qrq46f-0 EventForm__InputColumn2-sc-1hr7ssh-3 hKSNO ekNFdU"
        id="location"
        name="location"
        placeholder="location of the event"
      />
      <h2
        class="EventForm__EventInfosText-sc-1hr7ssh-8 jaKOTo"
      >
        EventInfos
      </h2>
      <label
        class="Label-sc-4r18cf-0 EventForm__InputLabelColumn1-sc-1hr7ssh-1 kTDDAh ckpfaq"
        for="EventStartDate"
      >
        Duration:
      </label>
      <div
        class="EventForm__EventStartDateDiv-sc-1hr7ssh-9 gZQGfC"
      >
        <div
          class="react-datepicker-wrapper"
        >
          <div
            class="react-datepicker__input-container"
          >
            <input
              class="Input-sc-1qrq46f-0 hKSNO"
              placeholder="start date"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        class="EventForm__EventEndDateDiv-sc-1hr7ssh-12 fPEBNm"
      >
        <div
          class="react-datepicker-wrapper"
        >
          <div
            class="react-datepicker__input-container"
          >
            <input
              class="Input-sc-1qrq46f-0 hKSNO"
              placeholder="end date"
              value=""
            />
          </div>
        </div>
      </div>
      <label
        class="Label-sc-4r18cf-0 EventForm__AddressInputLabel-sc-1hr7ssh-15 kTDDAh jwRQhJ"
        for="Street"
      >
        Address:
      </label>
      <input
        class="Input-sc-1qrq46f-0 EventForm__InputColumn2-sc-1hr7ssh-3 hKSNO dogpCS"
        id="street"
        name="street"
        placeholder="street + number"
      />
      <input
        class="Input-sc-1qrq46f-0 EventForm__InputColumn2-sc-1hr7ssh-3 hKSNO dogQeE"
        id="zip"
        name="zip"
        placeholder="zip"
      />
      <label
        class="Label-sc-4r18cf-0 EventForm__InputLabelColumn1-sc-1hr7ssh-1 kTDDAh eNdbsf"
        for="website"
      >
        Website:
      </label>
      <input
        class="Input-sc-1qrq46f-0 EventForm__InputColumn2-sc-1hr7ssh-3 hKSNO dofouy"
        id="website"
        name="website"
        placeholder="http://www.website.de"
      />
      <label
        class="Label-sc-4r18cf-0 EventForm__InputLabelColumn1-sc-1hr7ssh-1 kTDDAh eNdBTR"
        for="price"
      >
        TicketPrice:
      </label>
      <input
        class="Input-sc-1qrq46f-0 EventForm__InputColumn2-sc-1hr7ssh-3 hKSNO dofOWk"
        id="price"
        name="price"
        placeholder="ticket price or range"
      />
      <label
        class="Label-sc-4r18cf-0 EventForm__InputLabelColumn1-sc-1hr7ssh-1 kTDDAh eNdqlu"
        for="poster"
      >
        Picture:
         
      </label>
      <input
        class="Input-sc-1qrq46f-0 EventForm__InputColumn2-sc-1hr7ssh-3 hKSNO dojeif"
        id="poster"
        name="poster"
        placeholder="http://www.website.de/banner.jpg"
      />
      <label
        class="Label-sc-4r18cf-0 EventForm__InputLabelColumn1-sc-1hr7ssh-1 kTDDAh eNecAj"
        for="packlist"
      >
        Add PackList:
      </label>
      <select
        class="EventForm__PacklistInput-sc-1hr7ssh-16 bzsaqM"
        id="packlist"
        name="packlistCategory"
      >
        <option
          value="festival"
        >
          festival
        </option>
        <option
          value="medieval"
        >
          medieval
        </option>
      </select>
      <div
        class="EventForm__ButtonGroup-sc-1hr7ssh-17 oIMra"
      >
        <button
          type="reset"
        >
          Reset
        </button>
        <button
          class="EventForm__SubmitButton-sc-1hr7ssh-18 jeIQTA"
          type="submit"
        >
          Save
        </button>
      </div>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
